src/main/java/at/hannibal2/skyhanni/features/mining/crystalhollows/CrystalHollowsWalls.kt

```diff
@@ -0,0 +1,285 @@
+        draw(
+            LorenzVec(nucleusBBInflate.minX, nucleusBBInflate.minY, nucleusBBInflate.minZ),
+            LorenzVec(nucleusBBInflate.minX, nucleusBBInflate.minY, nucleusBBInflate.maxZ),
+            LorenzVec(nucleusBBInflate.maxX, nucleusBBInflate.minY, nucleusBBInflate.minZ),
+            Areas.HEAT.color
+        )
+        draw(
+            LorenzVec(nucleusBBInflate.minX, nucleusBBInflate.minY, nucleusBBInflate.minZ),
+            LorenzVec(nucleusBBInflate.minX, nucleusBBInflate.maxY, nucleusBBInflate.minZ),
+            LorenzVec(nucleusBBInflate.minX, nucleusBBInflate.minY, middleZ),
+            Areas.JUNGLE.color
+        )
+        draw(
+            LorenzVec(nucleusBBInflate.minX, nucleusBBInflate.minY, nucleusBBInflate.minZ),
+            LorenzVec(nucleusBBInflate.minX, nucleusBBInflate.maxY, nucleusBBInflate.minZ),
+            LorenzVec(middleX, nucleusBBInflate.minY, nucleusBBInflate.minZ),
+            Areas.JUNGLE.color
+        )
+        draw(
+            LorenzVec(nucleusBBInflate.maxX, nucleusBBInflate.minY, nucleusBBInflate.maxZ),
+            LorenzVec(nucleusBBInflate.maxX, nucleusBBInflate.maxY, nucleusBBInflate.maxZ),
+            LorenzVec(nucleusBBInflate.maxX, nucleusBBInflate.minY, middleZ),
+            Areas.PRECURSOR.color
+        )
+        draw(
+            LorenzVec(nucleusBBInflate.maxX, nucleusBBInflate.minY, nucleusBBInflate.maxZ),
+            LorenzVec(nucleusBBInflate.maxX, nucleusBBInflate.maxY, nucleusBBInflate.maxZ),
+            LorenzVec(middleX, nucleusBBInflate.minY, nucleusBBInflate.maxZ),
+            Areas.PRECURSOR.color
+        )
+        draw(
+            LorenzVec(nucleusBBInflate.minX, nucleusBBInflate.minY, nucleusBBInflate.maxZ),
+            LorenzVec(nucleusBBInflate.minX, nucleusBBInflate.maxY, nucleusBBInflate.maxZ),
+            LorenzVec(nucleusBBInflate.minX, nucleusBBInflate.minY, middleZ),
+            Areas.GOBLIN.color,
+        )
+        draw(
+            LorenzVec(nucleusBBInflate.minX, nucleusBBInflate.minY, nucleusBBInflate.maxZ),
+            LorenzVec(nucleusBBInflate.minX, nucleusBBInflate.maxY, nucleusBBInflate.maxZ),
+            LorenzVec(middleX, nucleusBBInflate.minY, nucleusBBInflate.maxZ),
+            Areas.GOBLIN.color
+        )
+        draw(
+            LorenzVec(nucleusBBInflate.maxX, nucleusBBInflate.minY, nucleusBBInflate.minZ),
+            LorenzVec(nucleusBBInflate.maxX, nucleusBBInflate.maxY, nucleusBBInflate.minZ),
+            LorenzVec(nucleusBBInflate.maxX, nucleusBBInflate.minY, middleZ),
+            Areas.MITHRIL.color
+        )
+        draw(
+            LorenzVec(nucleusBBInflate.maxX, nucleusBBInflate.minY, nucleusBBInflate.minZ),
+            LorenzVec(nucleusBBInflate.maxX, nucleusBBInflate.maxY, nucleusBBInflate.minZ),
+            LorenzVec(middleX, nucleusBBInflate.minY, nucleusBBInflate.minZ),
+            Areas.MITHRIL.color
+        )
```

Comment:
the first 2 lines of all 9 of these functions appears to be the same, maybe make a new function so that there is less repeated code

Reply:
nope, they are not the same, one value specific value changes for each of them. But I  think I can extract 4 meaningful points out of that.

Reply:
extracted what looked good